internal/api/handler/delete_project_v1alpha1.go

```diff
@@ -0,0 +1,54 @@
+package handler
+
+import (
+	"context"
+	"strings"
+
+	"github.com/bufbuild/connect-go"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	kubeerr "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/akuity/kargo/api/v1alpha1"
+	svcv1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+type DeleteProjectV1Alpha1Func func(
+	context.Context,
+	*connect.Request[svcv1alpha1.DeleteProjectRequest],
+) (*connect.Response[svcv1alpha1.DeleteProjectResponse], error)
+
+func DeleteProjectV1Alpha1(
+	kc client.Client,
+) DeleteProjectV1Alpha1Func {
+	return func(
+		ctx context.Context,
+		req *connect.Request[svcv1alpha1.DeleteProjectRequest],
+	) (*connect.Response[svcv1alpha1.DeleteProjectResponse], error) {
+		name := strings.TrimSpace(req.Msg.GetName())
+		if name == "" {
+			return nil, connect.NewError(connect.CodeInvalidArgument, errors.New("name should be empty"))
+		}
+
+		var ns corev1.Namespace
+		if err := kc.Get(ctx, client.ObjectKey{Name: name}, &ns); err != nil {
```

Comment:
Are there delete options that can take labels into account? If so, you might be able to get by without the get.